source/extensions/filters/http/fault/fault_filter.cc

```diff
@@ -129,34 +132,35 @@ Http::FilterHeadersStatus FaultFilter::decodeHeaders(Http::HeaderMap& headers, b
   return Http::FilterHeadersStatus::Continue;
 }
 
+bool FaultFilter::faultOverflow() {
+  const uint64_t max_faults = config_->runtime().snapshot().getInteger(
+      RuntimeKeys::get().MaxActiveFaultsKey, fault_settings_->maxActiveFaults().has_value()
+                                                 ? fault_settings_->maxActiveFaults().value()
+                                                 : std::numeric_limits<uint64_t>::max());
+  if (config_->stats().active_faults_.value() >= max_faults) {
+    config_->stats().faults_overflow_.inc();
+    return true;
+  } else {
+    return false;
+  }
+}
+
 bool FaultFilter::isDelayEnabled() {
-  bool enabled = config_->runtime().snapshot().featureEnabled(
-      DELAY_PERCENT_KEY, fault_settings_->delayPercentage().numerator(),
-      config_->randomGenerator().random(),
-      ProtobufPercentHelper::fractionalPercentDenominatorToInt(
-          fault_settings_->delayPercentage().denominator()));
+  bool enabled = config_->runtime().snapshot().featureEnabled(RuntimeKeys::get().DelayPercentKey,
+                                                              fault_settings_->delayPercentage());
```

Comment:
I am assuming the runtime code for featureEnabled(2 args) is equivalent to the prior one? I have little recollection of how the featureEnabled call got so many args..

Reply:
It's just because the newer version of the function didn't exist when this code was originally written. This just simplifies it, the behavior should be identical.